src/Parallel/Main.hpp

```diff
@@ -713,8 +722,8 @@ void Main<Metavariables>::execute_next_phase() {
   }
 
   if (Parallel::Phase::Exit == current_phase_) {
-    Informer::print_exit_info();
-    sys::exit();
+    did_all_elements_terminate(true);
```

Comment:
At first, I was a bit confused about why you were calling a reduction target of Main *on* Main. But then I realized that it won't actually do anything in `did_all_elements_terminate` except call the `contribute_termination_status_to_main()` entry on the chare corresponding to `current_termination_check_index_` (which should be 0 at this point).

How do you feel about adding a private member function to Main called `check_if_next_component_terminated_properly` that does
```cpp
auto* global_cache = Parallel::local_branch(global_cache_proxy_);
  ASSERT(global_cache != nullptr, "Could not retrieve the local global cache.");

  tmpl::for_each<component_list>([global_cache, this](auto component_tag_v) {
    using component_tag = tmpl::type_from<decltype(component_tag_v)>;
    if (tmpl::index_of<component_list, component_tag>::value ==
        current_termination_check_index_) {
      Parallel::get_parallel_component<component_tag>(*global_cache)
          .contribute_termination_status_to_main();
    }
  });
```

And calling that here and at the end of `did_all_elements_terminate` so you don't overuse a reduction target? You could even add an ASSERT/ERROR here that checks that `current_termination_check_index_ == 0`